R/utilities.R

```diff
@@ -4117,3 +4130,700 @@ supplement_maf <- function(incoming_maf,
+    )
+
+    ashm_matrix[ashm_matrix<=5] = 0
+    ashm_matrix[ashm_matrix>5] = 1
+
+
+    ashm_matrix <- ashm_matrix %>%
+        rename(
+            "PAX5_TSS" = "PAX5_TSS_1",
+            "SGK1_TSS" = "SGK1_TSS_1"
+    )
+
+    ashm_available <- colnames(ashm_matrix)
+
+    ashm_missing <- setdiff(
+        ashm_features,
+        ashm_available
+    )
+
+    if(length(ashm_missing)>0){
+        message(
+            "ATTENTION: Not all aSHM features are available in maf data!"
+        )
+        message(
+            paste0(
+                "A total of ",
+                length(
+                   ashm_missing
+                ),
+                " aSHM features are missing:"
+            )
+        )
+        message(
+            paste(
+                ashm_missing,
+                collapse=", "
+            )
+        )
+        message(
+            "They will be set to 0, which may affect model performance."
+        )
+        ashm_matrix[,ashm_missing] <- 0
+    }
+
+    # Combine together the SSM, hotspots, and aSHM
+    assembled_matrix <- bind_cols(
+            ssm_matrix,
+            ashm_matrix
+        ) %>%
+        select(
+            rownames(
+                model$importance
+            )
+        )
+
+    # Make prediction
+    prediction <- predict(
+        model,
+        assembled_matrix,
+        type="Vote"
+    )
+
+    prediction <- bind_cols(
+        prediction,
+        predict(
+                model,
+                assembled_matrix
+            ) %>%
+            as.data.frame %>%
+            `names<-`("is_cFL")
+        ) %>%
+    rownames_to_column("sample_id")
+
+    if(output=="predictions"){
+        return(prediction)
+    }else if (output=="matrix") {
+       return(assembled_matrix)
+    }else if(output=="both"){
+        return(
+            list(
+                predictions = prediction,
+                matrix = assembled_matrix)
+        )
+    }else{
+        stop("Invalid output type. Please specify predictions, matrix, or both.")
+    }
+
+
+}
+
+#' Classify DLBCLs according to genetic subgroups of Chapuy et al.
+#'
+#' Use the feature weights from NMF model to assemble the binary matrix and classify DLBCL tumors based on C0-C5 system of Chapuy et al
+#'
+#' @param these_samples_metadata The metadata data frame that contains sample_id column with ids for the samples to be classified.
+#' @param maf_data The MAF data frame to be used for matrix assembling. At least must contain the first 45 columns of standard MAF format.
+#' @param seg_data The SEG data frame to be used for matrix assembling. Must be of standard SEG formatting, for example, as returned by get_sample_cn_segments.
+#' @param sv_data The SV data frame to be used for matrix assembling. Must be of standard BEDPE formatting, for example, as returned by get_combined_sv.
+#' @param seq_type The seq_type of the samples. Only used to retrerive data through GAMBLR when it is not provided. Defaults to grch37.
```

Comment:
I think you got the `seq_type` and `projection` mixed up here based on the description. Hint, look at the defaults described for each parameter...

Reply:
Ah, thanks, will update!